src/m365/context/commands/context-init.spec.ts

```diff
@@ -0,0 +1,65 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import commands from '../commands';
+import { Logger } from '../../../cli/Logger';
+import { sinonUtil } from '../../../utils/sinonUtil';
+import appInsights from '../../../appInsights';
+import Command from '../../../Command';
+import * as fs from 'fs';
+const command: Command = require('./context-init');
+
+describe(commands.INIT, () => {
+  let log: any[];
+  let logger: Logger;
+
+  before(() => {
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      appInsights.trackEvent,
+      fs.existsSync,
+      fs.readFileSync
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      appInsights.trackEvent
+    ]);
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.INIT), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('writes a context to the .m365rc.json file', async () => {
```

Comment:
We should add here other tests for the same cases that we have in ContextCommand tests, but then calling the command rather than saveContext directly. While it doesn't matter for the coverage, it allows us to remove assumption of the implementation. Should we ever change the implementation, these tests will keep working and will help us ensure that everything is still working as expected.